releasenotes/notes/add-synth-mcx-with-ancillas-6a92078d6b0e1de4.yaml

```diff
@@ -0,0 +1,17 @@
+---
+features_synthesis:
+  - |
+    Add a synthesis function :func:`.synth_mcx_n_dirty_i15` that
+    synthesizes a multi-controlled X gate with :math:`k` controls using :math:`k - 2`
+    dirty ancillary qubits producing a circuit with at most :math:`8 * k - 6` CX gates,
+    by Iten et. al.
+  - |
+    Add a synthesis function :func:`.synth_mcx_n_clean_m15` that
+    synthesizes a multi-controlled X gate with :math:`k` controls using :math:`k - 2`
+    clean ancillary qubits producing a circuit with at most :math:`6 * k - 6` CX gates,
+    by Maslov.
```

Comment:
Do you think it's worth providing the full reference to the paper, the link to arxiv, or at least the name, i.e. instead of saying "by Maslov" say something like "as per the paper 'On the advantages of using relative phase Toffolis with an application to multiple control Toffoli optimization' by Maslov"?

Reply:
I'll add a reference to the arxiv.